src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/BaseTest.cs

```diff
@@ -55,6 +49,20 @@ static BaseTest ()
 			}
 		}
 
+		protected static string RunAdbCommand (string command, bool ignoreErrors = true)
+		{
+			string ext = Environment.OSVersion.Platform != PlatformID.Unix ? ".exe" : "";
+			var home = Environment.GetFolderPath (Environment.SpecialFolder.Personal);
+			string adb = Path.Combine (Environment.GetEnvironmentVariable ("ANDROID_SDK_PATH") ?? $"{home}/android-toolchain/sdk", "platform-tools", "adb" + ext);
+			var proc = System.Diagnostics.Process.Start (new System.Diagnostics.ProcessStartInfo (adb, command) { RedirectStandardOutput = true, RedirectStandardError = true, UseShellExecute = false });
+			proc.WaitForExit ();
+			var result = proc.ExitCode == 0 ?
+				proc.StandardOutput.ReadToEnd ().Trim () :
```

Comment:
Style: please place `?` and `:` at *beginning* of line, not end of line:

    var result = proc.ExitCode == 0
        ? proc.StandardOutput.ReadToEnd ().Trim ()
        : ignoreErrors ? String.Empty : proc.StandardError.ReadToEnd ().Trim ();

I'm also not sure I like the semantics of `RunAdbCommand()`: it returns stdout if there is no error, but stderr *or* the empty string if there is an error, with no way of knowing which one you got. Additionally, even when -- especially when! -- a program exits with an error, stdout may (will) still contain useful, meaningful, output.